### PR TITLE
Make vcs zip handling more robust.

### PR DESCRIPTION
**Pull request by jsirois**
This change works with both current vendored Pip and Pip 22.2.2.

Work towards #1805

### CODE REVIEW

**Comment by jsirois** [comment]
This is part 1 of 3 in breaking the final ~1k delta lines into chunks for review.

File: pex/pip/vcs.py [hunk 1]
```diff
@@ -50,7 +84,7 @@ def digest_vcs_archive(
     # `pip._internal.req.req_install.InstallRequirement.archive`.
     with TRACER.timed(
         "Digesting {archive} {vcs} archive".format(archive=os.path.basename(archive_path), vcs=vcs)
-    ), temporary_dir(cleanup=False) as chroot, open_zip(archive_path) as archive:
+    ), temporary_dir() as chroot, open_zip(archive_path) as archive:
```
Comment (Eric-Arellano) [inline pex/pip/vcs.py:87]: Why this change?

**Review by Eric-Arellano — APPROVED** [review]

File: pex/pip/vcs.py (hunk 1 above)
Comment (jsirois) [reply pex/pip/vcs.py:87]: Unrelated to this change to be sure, but a long standing lack of cleanup from some bug hunt in the past that I spent 15 minutes scrutinizing for, in the end, no reason. No more!
